### ISSUE TITLE
no-shadowed-variable should ignore shadowed `_`-params

### ISSUE THREAD

**Issue by ozra** [palantir/tslint#4201]
Having a few nested arrow functions that throw away the argument shouldn't be considered shadowing. Bad taste perhaps. But not shadowing.

**Comment by lukyth** [comment]
I agree with this. If this issue is accepting for PR, I'd like to take a shot at it.

**Comment by lukyth** [comment]
What about a case when `_` was used for [lodash](https://lodash.com); i.e. `import _ from 'lodash'`? `_` in function argument would probably be considered as shadowed in that case.

An option to allow some variable name to be shadowed like [what ESLint does with its `no-shadow` rule](https://eslint.org/docs/rules/no-shadow#allow) might be better. Then shadowed `_` can be allowed if `lodash` is not being used.

**Comment by tao-cumplido** [comment]
Just came across this myself when I noticed that typescripts lets me use `_` as argument name with compiler flag `noUnusedParameters` enabled.

I then stumbled upon [this issue](https://github.com/Microsoft/TypeScript/issues/9458) which states that any name starting with `_` is allowed and suggests using names like `_1`, `_2`, etc. for multiple unused parameters. The same approach can be used to prevent the linting rule from triggering.
